src/simulator/simulator.py

```diff
@@ -6,44 +6,76 @@
 WORLD_HEIGHT = 160
 
 import pygame
+from copy import deepcopy
 import numpy as np
 from classes.vehicle import Vehicle
+from classes.button import Button
 from manager.manager import Manager
 from manager.route import Node, Edge, Route
 from .render import render_world, render_manager
 from .update import update_world
 
+def vehicle_copy(vehicles: list[Vehicle]) -> list[Vehicle]:
+    return [Vehicle(v.id, v.position, v.velocity, v.acceleration, v.direction, v.width, v.length, v.pivot_distance, v.intent, v.image_source) for v in vehicles]
+
 def run_simulation(initial_vehicles: list[Vehicle], nodes: list[Node], edges: list[Edge], routes: list[Route], manager: Manager): # requires initialization of lanes, manager, vehicles
     pygame.init()
     screen = pygame.display.set_mode((SCREEN_WIDTH, SCREEN_HEIGHT))
     clock = pygame.time.Clock()
     running = True
-    delta_time: float = 0
+    delta_time = 0
+
+    vehicles = vehicle_copy(initial_vehicles)
+    to_update = True
```

Comment:
is it possible to use lambda functions for this? am wondering if run_simulation can be as readable as possible. Also maybe change the name of to_update to maybe is_paused

Reply:
I'll look into this, I think by nature pygame doesn't present a lot of clean coding practices.